conan/cli/commands/list.py

```diff
@@ -102,6 +102,8 @@ def list(conan_api: ConanAPI, parser, *args):
     parser.add_argument('-p', '--package-query', default=None, action=OnceArgument,
                         help="List only the packages matching a specific query, e.g, os=Windows AND "
                              "(arch=x86 OR compiler=gcc)")
+    parser.add_argument('-pr', '--profile', default=None, action="append",
+                        help="Profiles to filter the package binaries")
```

Comment:
I'm not sure if this param name is clear enough to say "filtering packages by this profile". We already have the `-p` param so I think it could be a bit confusing.